url/url.cpp

```diff
@@ -191,6 +191,74 @@ std::string blob_url_create(Origin const &origin) {
     return result;
 }
 
+// https://url.spec.whatwg.org/#concept-host-serializer
+std::string Host::serialize() const {
+    if (type == HostType::Ip4Addr) {
+        return util::ipv4_serialize(std::get<std::uint32_t>(data));
+    } else if (type == HostType::Ip6Addr) {
+        return util::ipv6_serialize(const_cast<std::array<std::uint16_t, 8> &>(std::get<2>(data)));
+    }
+
+    return std::get<std::string>(data);
+}
+
+// https://url.spec.whatwg.org/#url-path-serializer
+std::string Url::serialize_path() const {
+    if (std::holds_alternative<std::string>(path)) {
```

Comment:
You can check if a variant holds a specific type and get a pointer to it at the same time using `std::get_if`. If the pointer != nullptr, you have an object of that type.